src/main/java/uk/co/jemos/podam/api/AbstractRandomDataProviderStrategy.java

```diff
@@ -542,8 +542,17 @@ public RandomDataProviderStrategy removeAttributeStrategy(
 	public AttributeStrategy<?> getStrategyForAnnotation(
 			final Class<? extends Annotation> annotationClass) {
 
-		return attributeStrategies.get(annotationClass);
+		if (attributeStrategies.containsKey(annotationClass)) {
+			return attributeStrategies.get(annotationClass);
+		} else {
+			for (Entry<Class<? extends Annotation>, AttributeStrategy<?>> attributeStrategy: attributeStrategies.entrySet()) {
+				if (attributeStrategy.getKey().isAssignableFrom(annotationClass)) {

```

Comment:
Now order in which map is iterated is defining the result. attributeStratelegies is a hash map, meaning the result will be unpredictable.